ffi/src/lib.rs

```diff
@@ -29,7 +30,10 @@ use handle::Handle;
 // relies on `crate::`
 extern crate self as delta_kernel_ffi;
 
+pub mod engine_data;
 pub mod engine_funcs;
+pub mod error;
+pub use error::*;
```

Comment:
Is there a reason to bring all the error stuff into the root scope? It's a breaking change if we don't, but also, I think we _do_ want things to live in `error`? It's also consistent with the `kernel` crate where we don't bring everything from `error` into the root.

Reply:
ok sounds good - I was thinking it would be 'better' to have this not change the APIs but I do agree it probably just makes sense to have them be exposed inside of `mod error` - ill do that!